scripts/globals/job_utils/dancer.lua

```diff
@@ -181,23 +181,28 @@ xi.job_utils.dancer.checkWaltzAbility = function(player, target, ability)
     elseif player:getTP() < waltzInfo[1] then
         return xi.msg.basic.NOT_ENOUGH_TP, 0
     else
-        -- Waltz Delay (-1s per mod value)
+        local newRecast = ability:getRecast()
+
+        -- Apply Waltz Delay Modifier (-1s per mod value)
         local recastMod = player:getMod(xi.mod.WALTZ_DELAY)
+
         if recastMod ~= 0 then
-            local newRecast = ability:getRecast() + recastMod
-            ability:setRecast(utils.clamp(newRecast, 0, newRecast))
+            newRecast = newRecast + recastMod
         end
 
         -- Apply "Fan Dance" Waltz recast reduction.  All tiers above 1 grant 5%
         -- recast reduction each.
-        local fanDanceMerits = target:getMerit(xi.merit.FAN_DANCE)
+        local fanDanceMeritValue = target:getMerit(xi.merit.FAN_DANCE)
+
         if
             player:hasStatusEffect(xi.effect.FAN_DANCE) and
-            fanDanceMerits > 1
+            fanDanceMeritValue > 5 -- 1 merit = Value of 5.
```

Comment:
This is changing the logic completely for applying merit changes by themself.  This reduction should apply for every merit past the first, at a 5% reduction

Reply:
I see now.  Could you please comment the fandDanceMeritValue variable as well to understand that it is receiving a value of 5 for each upgrade?

Reply:
Done